### PR TITLE
OCPBUGS-3404: operator/ingress: matchExpressions cannot be used

### PR DESCRIPTION
**Pull request by Miciah**
Document that `nodePlacement.matchExpressions` cannot be used, due to a limitation in the Kubernetes pod API.

* `operator/v1/types_ingress.go` (`NodePlacement`): Document that `nodePlacement.matchExpressions` cannot be used.
* `openapi/generated_openapi/zz_generated.openapi.go`:
* `openapi/openapi.json`:
* `operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml`:
* `operator/v1/zz_generated.swagger_doc_generated.go`: Regenerate.  

---

Follow-up to https://github.com/openshift/api/pull/870#discussion_r601577395.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
/lgtm

Thanks for keeping the docs up to date

**Comment by Miciah** [comment]
Last push just replaces the Bugzilla tracker with a Jira tracker so that I can get the "valid-bug" labels.

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -319,6 +319,11 @@ type NodePlacement struct {
 	//
 	// These defaults are subject to change.
 	//
+	// Note that using nodeSelector.matchExpressions is not supported.  Only
+	// nodeSelector.matchLabels may be used.  This is a limitation of the
+	// Kubernetes API: the pod spec does not allow complex expressions for
```
Comment (frobware) [inline operator/v1/types_ingress.go:324]: Is there an upstream issue we could reference either here or in the PR description?

**Pull request merged**